### PR TITLE
New simplified User ReadPool related changes

### PR DESCRIPTION
**Pull request by iamrz1**
### Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/blobber/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

### Associated PRs (Link as appropriate):
- 0chain: https://github.com/0chain/0chain/pull/1214
- gosdk: https://github.com/0chain/gosdk/pull/461
- system_test: https://github.com/0chain/system_test/pull/277
- zboxcli: https://github.com/0chain/zboxcli/pull/234
- zwalletcli:
- Other: ...

### CODE REVIEW

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | user-read-pools  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | user-read-pools  |
| zwalletcli  | staging  |
| 0chain  | user-read-pools  |
| blobber  | user-read-pools  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | user-read-pools  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | user-read-pools  |
| zwalletcli  | user-read-pools  |
| 0chain  | user-read-pools  |
| blobber  | user-read-pools  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | user-read-pools  |
| zwalletcli  | staging  |
| 0chain  | user-read-pools  |
| blobber  | user-read-pools  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | user-read-pools  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | user-read-pools  |
| zwalletcli  | staging  |
| 0chain  | user-read-pools  |
| blobber  | user-read-pools  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

File: code/go/0chain.net/blobbercore/allocation/entity.go [hunk 1]
```diff
@@ -302,18 +290,3 @@ type ReadPoolRedeem struct {
 	PoolID  string `json:"pool_id"` // read pool ID
 	Balance int64  `json:"balance"` // balance reduction
 }
-
-// SubReadRedeemed subtracts tokens redeemed from read pools.
-func SubReadRedeemed(rps []*ReadPool, redeems []ReadPoolRedeem) {
```
Comment (dabasov) [inline code/go/0chain.net/blobbercore/allocation/entity.go:307]: why do you remove redeem logic?

Comment (iamrz1) [reply code/go/0chain.net/blobbercore/allocation/entity.go:307]: it was a tricky one, I just remembered. Basically we didn't have bunch of pool ids anymore, so the information here is no goo to us. So my theory is we could probably update read pool by aggregating the the cost here and then subtracting from appropriate account in read-pool table. But I didn't end up doing it as things seems to have been working as is.
We still update the existing read pool with new values from the chain in the method `(rm *ReadMarkerEntity) Sync(ctx context.Context)`

**Review by dabasov — APPROVED** [review]
lgtm

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | TRUE  |
| system-tests  | master  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | user-read-pools  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Pull request merged**